packages/contracts/contracts/voting/admin-address/AdminAddressSetup.sol

```diff
@@ -0,0 +1,120 @@
+/// @notice The setup contract of the `AdminAddress` plugin.
+contract AdminAddressSetup is PluginSetup {
+    using Clones for address;
+
+    /// @notice The address of `AdminAddress` plugin logic contract to be cloned.
+    address private immutable adminAddressBase;
+
+    /// @notice The address zero to be used as oracle address for permissions.
+    address private constant NO_ORACLE = address(0);
+
+    /// @notice Thrown if admin address is zero.
+    /// @param admin The admin address.
+    error AdminAddressInvalid(address admin);
+
+    /// @notice The contract constructor, that deployes the `AdminAddress` plugin logic contract.
+    constructor() {
+        adminAddressBase = address(new AdminAddress());
+    }
+
+    /// @inheritdoc IPluginSetup
+    function prepareInstallationDataABI() external pure returns (string memory) {
+        return "(address admin)";
+    }
+
+    /// @inheritdoc IPluginSetup
+    function prepareInstallation(address _dao, bytes memory _data)
+        external
+        returns (
+            address plugin,
+            address[] memory helpers,
+            PermissionLib.ItemMultiTarget[] memory permissions
+        )
+    {
+        IDAO dao = IDAO(_dao);
+
+        // Decode `_data` to extract the params needed for cloning and initializing `AdminAddress` plugin.
+        address admin = abi.decode(_data, (address));
+
+        if (admin == address(0)) {
+            revert AdminAddressInvalid({admin: admin});
+        }
+
+        // Clone plugin contract.
+        plugin = adminAddressBase.clone();
+
+        // Initialize cloned plugin contract.
+        AdminAddress(plugin).initialize(dao);
+
+        // Prepare helpers
+        helpers = new address[](0);
+
+        // Prepare permissions
+        permissions = new PermissionLib.ItemMultiTarget[](2);
+
+        // Grant `ADMIN_EXECUTE_PERMISSION` of the Plugin to the admin.
+        permissions[0] = PermissionLib.ItemMultiTarget(
+            PermissionLib.Operation.Grant,
+            plugin,
+            admin,
+            NO_ORACLE,
+            AdminAddress(plugin).ADMIN_EXECUTE_PERMISSION_ID()
+        );
+
+        // Grant `EXECUTE_PERMISSION` of the DAO to the plugin.
+        permissions[1] = PermissionLib.ItemMultiTarget(
+            PermissionLib.Operation.Grant,
+            _dao,
+            plugin,
+            NO_ORACLE,
+            DAO(payable(_dao)).EXECUTE_PERMISSION_ID()
+        );
+    }
+
+    /// @inheritdoc IPluginSetup
+    function prepareUninstallationDataABI() external pure returns (string memory) {
+        return "";
+    }
+
+    /// @inheritdoc IPluginSetup
+    /// @dev Currently there is not a relaiable mean to revoke plugin's permissions such as `ADMIN_EXECUTE_PERMISSION_ID`
+    /// that have been granted to addresses during the life cycle of the plugin.
+    /// or the ones that have been granted are not revoked already,
+    /// therefore, only `EXECUTE_PERMISSION_ID` is revoked for this uninstallation.
```

Comment:
I wouldn't include this text in a plugin that is published.

Why did you revert from having a state variable the `address admin` and setting it via the initializer so that you can query it during the uninstallation?

Reply:
No, it's not reliable and is a wrong way of doing what we want to do. I understand why you want to have it, but it's just a wrong way of doing this ! This also happens in other cases, which is why we need to discuss this thing ! Let's do it today at some point.